pandas/core/dtypes/common.py

```diff
@@ -1236,7 +1236,8 @@ def is_datetime_or_timedelta_dtype(arr_or_dtype):
     if arr_or_dtype is None:
         return False
     tipo = _get_dtype_type(arr_or_dtype)
-    return issubclass(tipo, (np.datetime64, np.timedelta64))
+    return (issubclass(tipo, (np.datetime64, np.timedelta64)) or
+            is_datetime64tz_dtype(arr_or_dtype))
```

Comment:
Is this function used often? There is a lot of overlap with similar functions here

Reply:
it has a few references. would not object to removing it (it is used in nanops in determining whether to use bottleneck, which was the problem here)